pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -74,10 +77,19 @@ type ResourceQuotaControllerOptions struct {
 	IgnoredResourcesFunc func() map[schema.GroupResource]struct{}
 	// InformersStarted knows if informers were started.
 	InformersStarted <-chan struct{}
-	// InformerFactory interfaces with informers.
-	InformerFactory InformerFactory
+	// SharedInformerFactory interfaces with shared informers.
+	SharedInformerFactory informers.SharedInformerFactory
 	// Controls full resync of objects monitored for replenishment.
 	ReplenishmentResyncPeriod controller.ResyncPeriodFunc
+	// dynamicClient is used to work with unknown resources,
+	// like those created via CRDs or aggregated apiservers.
+	DynamicClient dynamic.Interface
+	// RESTMapper can reset itself from discovery.
+	RESTMapper resettableRESTMapper
```

Comment:
Is the underlying impl for this threadsafe?  We use a generally dynamic restmapper in the controller context, but I don't know if it is designed to tolerate concurrent resets.

Reply:
Yes.

https://github.com/kubernetes/kubernetes/blob/28f171bd66937dec8b24a05c4b7a1414432f9fe8/staging/src/k8s.io/client-go/restmapper/discovery.go#L210-L220